specs/rollup-node.md

```diff
@@ -59,31 +64,82 @@ currently only concerned with the specification of the rollup driver.
+
+Each sequencing window is derived into a variable number of L2 blocks, depending on the timestamps of L1 and L2.
+
+The L2 has a fixed block time and no more than one batch per block,
+meaning that gaps between the batches (ordered by timestamp) are interpreted as batches with empty transaction-lists,
+thus construing L2 blocks that only contain deposit transaction(s).
+
+The L2 blocks produced by a sequencing window are bounded by timestamp:
+
+- `min_l2_timestamp = prev_l2_timestamp + l2_block_time`
+- `max_l2_timestamp = l1_timestamp + l2_block_time`, where `l1_timestamp` is the timestamp of the
+  first L1 block of the sequencing window. (maximum bound, may not be aligned with block time)
+
+If there are no batches present in the sequencing window then the L2 chain is extended up to `max_l2_timestamp` (incl.)
+with empty batches, but otherwise regular block derivation.
+
+Note that with short block times on L1 the L2 time may increment beyond the L1 time,
+but the longer target block time of L1 will correct back and allow the timestamps to align again.
+
 ## From L1 Blocks to Payload Attributes
 
 ### Reading L1 inputs
 
-The rollup reads the following data from each L1 block:
-
-- L1 block attributes
-  - block number
-  - timestamp
-  - basefee
-  - *random* (the output of the [`RANDOM` opcode][random])
-- L1 log entries emitted for [user deposits][g-deposits], augmented with
-  - `blockHeight`: the block-height of the L1 block
-  - `transactionIndex`: the transaction-index within the L2 transactions list
+The rollup reads the following data from the [sequencing window][g-sequencing-window]:
+
+- Of the *first* block in the window only:
+  - L1 block attributes:
+    - block number
+    - timestamp
+    - basefee
+    - *random* (the output of the [`RANDOM` opcode][random])
+  - L1 log entries emitted for [user deposits][g-deposits], augmented with
+    - `blockHeight`: the block-height of the L1 block
+    - `transactionIndex`: the transaction-index within the L2 transactions list
+- Of each block in the window:
+  - Sequencer batches, derived from the transactions:
+    - The transaction receiver is the sequencer inbox address
+    - The transaction must be signed by a recognized sequencer account
+    - The calldata may contain any number of batches. *(calldata will be substituted with blob data in the future.)*
+    - Batches not matching filter criteria are ignored:
+      - `batch.epoch == sequencing_window.epoch`, i.e. for this sequencing window
+      - `(batch.timestamp - genesis_l2_timestamp) % block_time == 0`, i.e. timestamp is aligned
+      - `min_l2_timestamp < batch.timestamp < max_l2_timestamp`, i.e. timestamp is within range
+      - The batch is the first batch with `batch.timestamp` in this sequencing window,
+        i.e. one batch per L2 block number
 
 [random]: https://eips.ethereum.org/EIPS/eip-4399
 
+Batches are formatted as: `type ++ RLP([epoch, timestamp, transaction_list])`
```

Comment:
When being abstract, should it be `type ++ blob` where type 0 is `type ++ RLP([epoch, timestamp, transaction_list])` ?

Reply:
Updated this to word it better and also include encoding of multiple batches (i.e. when we are submitting multiple L2 blocks together)